docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,181 @@
+* Routing, e.g. PROXY protocol, original destination, dynamic forwarding.
+* Authentication and authorization.
+* Circuit breakers and outlier detection.
+* Many other networking, HTTP, listener, access logging, health checking, tracing and stats
+  extensions and configuration options.
+
+It's helpful to limit the nature of the configuration and request to provide an illustrative
+narrative below, so we opt for the following parameters:
+
+* An HTTP/2 request with TLS over a TCP connection for both downstream and upstream.
+* The HTTP connection manager as the only network filter.
+* The router filter as the only HTTP filter (TODO: should we add more filters?).
+* Filesystem access logging.
+* Statsd sink.
+
+We assume a static bootstrap configuration, but this is a control plane detail that is
+orthogonal to the data plane:
+
+.. code-block:: yaml
+
+  todo: simple bootstrap example of above
+
+High level architecture
+-----------------------
+
+The request processing path in Envoy has responsibilities split across two major loosely
+coupled subsystems:
+
+* *ListenerManager* where the downstream client related aspects of request processing take place.
+  This includes all processing from when the first bytes are read from the socket until a
+  decoded HTTP request is ready to forward to an upstream cluster. This subsystem is also
+  responsible for the response path to the client, the downstream HTTP/2 codec lives here.
+* *ClusterManager* where the HTTP request is forwarded to an upstream backend. This is where
+  knowledge of cluster and endpoint health, load balancing and connection pooling exists. The
+  upstream HTTP/2 codec lives here.
+
+The two subsystems are bridged with the HTTP router filter.
+
+We use the labels *ListenerManager* and *ClusterManager* above to refer to the family of modules and
+instance classes that are rooted in these top-level manager classes. In reality, there are numerous
+components that we discuss below that are instantiated prior to and during the course of a request
+by these management systems, for example listeners, filter chains, codecs, connection pools and load
+balancing data structures.
+
+.. note::
+  TODO: add architecture diagram.
+
+Request path
+------------
+
+Overview
+~~~~~~~~
+
+The request path for the initial bytes that constitute the request headers takes the following
+steps:
+
+1. A TCP connection from a downstream is accepted by an Envoy listener associated with a worker
+   thread.
+2. The listener filter chain is instantiated and executed, which can provide SNI and other pre-TLS info.
+3. On network reads, the TLS transport socket transforms the encrypted TCP connection
+   to a plain text network connection.
+4. The network filter chain is instantiated and executed. The most significant filter for HTTP is
+   the HTTP connection manager which sits at the end of the filter chain.
+5. The HTTP/2 codec in HTTP connection manager lifts the plain text TCP connection to a number of
```

Comment:
lifts -> ???
for H2 enthusiastis I'd use demuxes but that's probably not going to clear things up for non-H2 geeks.  transforms?  turns the TCP connection into ...?